Card constructor signature change, remove test package, finish IO

secminhr opened the pull request:
And some fixes also.

secminhr committed:
Fix logic error in game.

secminhr committed:
Change method name in EventManager.
Finish Game.teardown().

secminhr committed:
merging

secminhr committed:
Merge remote-tracking branch 'upstream/master'

secminhr committed:
Fix Sundar's condition problem, which responsible for its third passive skill.

secminhr committed:
Change Card constructor signature.
Finish LocalIO, IO interface.
Finish GameStarter.

secminhr committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	src/main/java/eod/Player.java
#	src/main/java/eod/card/abstraction/summon/SummonCard.java
#	src/main/java/eod/card/concrete/summon/DroneSummon.java
#	src/main/java/eod/card/concrete/summon/GhostOfHatredSummon.java
#	src/main/java/eod/card/concrete/summon/GuerrillaShooterSummon.java
#	src/main/java/eod/card/concrete/summon/LeadersGuardSummon.java
#	src/main/java/eod/card/concrete/summon/LittleGhostSummon.java
#	src/main/java/eod/card/concrete/summon/SecureGuardBotSummon.java
#	src/main/java/eod/card/concrete/summon/SpacezipperSummon.java

secminhr committed:
Fix bugs caused by Card constructor signature change.
Delete Test package, since it's not working.
Change attacker in SecureGuardBot to private.

secminhr committed:
Bug fix.

secminhr committed:
Merge remote-tracking branch 'upstream/master'

secminhr committed:
Fix cards' constructor signature problem.

aoaaceai commented on src/main/java/eod/IO/Input.java:
```diff
@@ -1,5 +1,20 @@
+
+    //This method will send requesting message to user, waiting for its response
+    //and return the desired card.
+    //Return null if player want to ended "Action stage(行動階段的英文不會)".
```
how about Action Phase?

secminhr replied:
Changed.

aoaaceai commented on src/main/java/eod/Player.java:
```diff
@@ -72,13 +90,18 @@ public void dropCards() {
+
+    public void startAutoAttackInOrder() {
+        Gameboard board = game.getBoard();
+
```
// TodO

secminhr replied:
added

secminhr replied:
And will be finished in next pr.

aoaaceai commented on src/main/java/eod/IO/Input.java:
```diff
@@ -1,5 +1,20 @@
+
+import java.awt.*;
+
 public interface Input {
```
todo: connection with Player.selectObject, selectPosition, and selectCard
or maybe remove those methods.

secminhr replied:
Added

aoaaceai commented on src/main/java/eod/effect/Heal.java:
```diff
@@ -14,7 +14,7 @@ public Heal(Player player, int hp) {
     }
 
-    public void from(WarObject[] objects) {
+    public void to(WarObject[] objects) {
```
// tOdO: add to(WarObject[] objects, int num) to heal multiple characters.

secminhr replied:
Added

secminhr committed:
Use Action Phase.
TODO in Player.startAutoAttackInOrder() added, and will be finished in next pr.
Fixing lack of required ability of Input interface.
Added Heal.to(objects, number).
